backend/src/main/java/org/atlantbh/internship/auctionapp/service/impl/BidServiceImpl.java

```diff
@@ -26,7 +37,26 @@ public ProductBidResponse getProductBidInfo(Long productId) {
     }
 
     @Override
-    public List<UserProfileProductsInfo> getBidsForUser(Long userId) {
+    public List<ProductBidsInfo> getBidsForUser(Long userId) {
         return bidRepository.getUserBids(userId);
     }
+
+    @Override
+    public SuccessfulAndMessage bid(BidRequest bidRequest, PersonDetails user) throws BadRequestException {
+        ProductEntity product = productRepository.findById(bidRequest.getProductId())
+                .orElseThrow(() -> new BadRequestException("Product with given id does not exist."));
+        if (product.getUser().getId().equals(user.getId()))
+            throw new BadRequestException("You can't bid to your own products");
+        BigDecimal currentHighest = bidRepository.findFirst1ByProductIdOrderByPriceDesc(bidRequest.getProductId())
+                .map(BidEntity::getPrice).orElse(product.getStartingPrice());
+        if (product.getEndDate().isBefore(LocalDateTime.now()))
+            return new SuccessfulAndMessage(false, "You can no longer bid on this product");
+        if (bidRequest.getPrice().compareTo(product.getStartingPrice()) <= 0)
+            return new SuccessfulAndMessage(false, "You have to provide value grater than starting price.");
+        if (bidRequest.getPrice().compareTo(currentHighest) <= 0)
+            return new SuccessfulAndMessage(false, "There are higher bids than yours. You could give a second try.");
```

Comment:
I believe this should be `<` instead of `<=` when comparing the `currentHighest`?